webapi/src/main/scala/org/knora/webapi/slice/admin/api/model/ProjectsEndpointsRequestsAndResponses.scala

```diff
@@ -42,13 +45,33 @@ object ProjectsEndpointsRequestsAndResponses {
     implicit val codec: JsonCodec[ProjectUpdateRequest] = DeriveJsonCodec.gen[ProjectUpdateRequest]
   }
 
-  final case class SetRestrictedViewRequest(size: RestrictedViewSize, watermark: Option[Boolean])
+  final case class SetRestrictedViewRequest(
+    size: Option[RestrictedView.Size],
+    watermark: Option[RestrictedView.Watermark]
+  ) {
+    def toRestrictedView: IO[BadRequestException, RestrictedView] =
+      (size, watermark) match {
+        case (Some(size), None)      => ZIO.succeed(size)
+        case (None, Some(watermark)) => ZIO.succeed(RestrictedView.Watermark.from(watermark.value))
+        case _                       => ZIO.fail(BadRequestException("Exactly one one of size or watermark must be provided."))
```

Comment:
```suggestion
  final case class SetRestrictedViewRequest(
    size: Option[RestrictedView.Size],
    watermark: Option[RestrictedView.Watermark]
  ) {
    def toRestrictedView: IO[BadRequestException, RestrictedView] =
      (size, watermark) match {
        case (Some(size), None)      => ZIO.succeed(size)
        case (None, Some(watermark)) => ZIO.succeed(RestrictedView.Watermark.from(watermark.value))
        case _                       => ZIO.fail(BadRequestException("It is only possible to set either the size or the watermark, not both at the same time."))
```
If you would like to keep old error message, one `one` should be removed.

Reply:
For me the first part `Specify either the size or the watermark.` is already enough, but take what do you feel best.